akka-actor-typed/src/main/scala/akka/actor/typed/Behavior.scala

```diff
@@ -196,6 +196,12 @@ object Behavior {
     override def toString = "Unhandled"
   }
 
+  /**
+   * INTERNAL API
+   */
+  @InternalApi
+  private[akka] def failed[T](cause: Throwable): Behavior[T] = new FailedBehavior(cause).asInstanceOf[Behavior[T]]
```

Comment:
It'd be nice if this (and his mates in this file) used an `unsafeUpcast` (which doesn't exist on `Behavior`) rather than raw `asInstanceOf`. cf. #25896

Reply:
Totally feature-creeping your PR, I realised after the fact.  Just a thought.

Reply:
Yeah we have it for actors

Reply:
@dwijnand it would be v nice.

Reply:
Fixed in https://github.com/akka/akka/pull/26093 (cc @hepin1989)